mono/metadata/process.c

```diff
@@ -323,57 +333,38 @@ process_get_fileversion (MonoObject *filever, gunichar2 *filename)
 					 * otherwise
 					 */
 					lang_count = VerLanguageName (lang & 0xFFFF, lang_buf, 128);
-					if (lang_count)
-						process_set_field_string (filever, "language", lang_buf, lang_count);
-					process_module_stringtable (filever, data, trans_data[0], trans_data[1]);
+					if (lang_count) {
+						process_set_field_string (filever, "language", lang_buf, lang_count, error);
+						if (!mono_error_ok (error))
+							return;
+					}
+					process_module_stringtable (filever, data, trans_data[0], trans_data[1], error);
+					if (!mono_error_ok (error))
+						return;
 				}
 			} else {
-				/* No strings, so set every field to
-				 * the empty string
-				 */
-				process_set_field_string (filever,
-							  "comments",
-							  EMPTY_STRING, 0);
-				process_set_field_string (filever,
-							  "companyname",
-							  EMPTY_STRING, 0);
-				process_set_field_string (filever,
-							  "filedescription",
-							  EMPTY_STRING, 0);
-				process_set_field_string (filever,
-							  "fileversion",
-							  EMPTY_STRING, 0);
-				process_set_field_string (filever,
-							  "internalname",
-							  EMPTY_STRING, 0);
-				process_set_field_string (filever,
-							  "legalcopyright",
-							  EMPTY_STRING, 0);
-				process_set_field_string (filever,
-							  "legaltrademarks",
-							  EMPTY_STRING, 0);
-				process_set_field_string (filever,
-							  "originalfilename",
-							  EMPTY_STRING, 0);
-				process_set_field_string (filever,
-							  "privatebuild",
-							  EMPTY_STRING, 0);
-				process_set_field_string (filever,
-							  "productname",
-							  EMPTY_STRING, 0);
-				process_set_field_string (filever,
-							  "productversion",
-							  EMPTY_STRING, 0);
-				process_set_field_string (filever,
-							  "specialbuild",
-							  EMPTY_STRING, 0);
+				int i;
+
+				for (i = 0; i < G_N_ELEMENTS (stringtable_entries); ++i) {
+					/* No strings, so set every field to
+					 * the empty string
+					 */
+					process_set_field_string (filever,
+											  stringtable_entries [i].name,
+											  EMPTY_STRING, 0, error);
+					if (!mono_error_ok (error))
+						return;
```

Comment:
There's a `return_if_nok(error)` macro that expands to exactly `if (!mono_error_ok (error)) return`, if you want.